apps/arweave/src/ar_mining_server.erl

```diff
@@ -639,23 +638,18 @@ prepare_solution(proofs, Candidate, Solution) ->
 
 prepare_solution(poa1, Candidate, Solution) ->
 	#mining_solution{
-		poa1 = CurrentPoA1,
-		packing_difficulty = PackingDifficulty,
-		mining_address = MiningAddress, partition_number = PartitionNumber,
-		recall_byte1 = RecallByte1 } = Solution,
-	#mining_candidate{
-		chunk1 = Chunk1, h0 = H0, nonce = Nonce,
-		packing_difficulty = PackingDifficulty,
-		partition_upper_bound = PartitionUpperBound } = Candidate,
-	case is_poa_complete(CurrentPoA1, PackingDifficulty) of
-		true ->
-			Solution;
-		false ->
-			Packing = ar_block:get_packing(PackingDifficulty, MiningAddress),
-			case {Chunk1, read_poa(RecallByte1, Chunk1, Packing, Nonce)} of
-				{_Chunk1, {ok, PoA1}} ->
-					Solution#mining_solution{ poa1 = PoA1 };
-				{not_set, {error, Error}} ->
+		poa1 = CurrentPoA1, recall_byte1 = RecallByte1,
+		mining_address = MiningAddress, packing_difficulty = PackingDifficulty
+	} = Solution,
+	#mining_candidate{ chunk1 = Chunk1, nonce = Nonce } = Candidate,
+
+	case prepare_poa(poa1, Candidate, CurrentPoA1) of
+		{ok, PoA1} ->
+			Solution#mining_solution{ poa1 = PoA1 };
+		{error, Error} ->
+			case Chunk1 of
+				not_set ->
+					Packing = ar_block:get_packing(PackingDifficulty, MiningAddress),
```

Comment:
The PR did not introduce it (I did [here](https://github.com/ArweaveTeam/arweave/commit/18e7eacb731781007ae9efb299d67c21f4923bb7#diff-ea8e957c63acb7497b0ed4f9201b40f18e5d5b4fa371e67b4cff05850e4df854R654),) but it does not look quite right - we should wrap the `Solution#mining_solution{ poa1 = #poa{ chunk = SubChunk } }` below in `may_be_leave_it_to_exit_peer` to make sure it errors our on exit peer because there are no proofs, only the raw chunk. Should we fix it here?

Reply:
Like this?

https://github.com/ArweaveTeam/arweave/pull/628/commits/eaeffd695b764623f046013b71e49034bd8f87cf

Reply:
Yes, thank you!